experiments/AtmosGCM/heldsuarez-tracers.jl

```diff
@@ -0,0 +1,273 @@
+using ClimateMachine.Thermodynamics:
+    air_temperature, internal_energy, air_pressure
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.VariableTemplates
+
+using Distributions: Uniform
+using LinearAlgebra
+using StaticArrays
+using Random: rand
+using Test
+
+using CLIMAParameters
+using CLIMAParameters.Planet: R_d, day, grav, cp_d, cv_d, planet_radius
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+struct HeldSuarezDataConfig{FT}
+    T_ref::FT
+end
+
+function init_heldsuarez!(bl, state, aux, coords, t)
+    FT = eltype(state)
+
+    # Set initial state to reference state with random perturbation
+    rnd = FT(1.0 + rand(Uniform(-1e-3, 1e-3)))
+    state.ρ = aux.ref_state.ρ
+    state.ρu = SVector{3, FT}(0, 0, 0)
+    state.ρe = rnd * aux.ref_state.ρe
+
+    if number_of_tracers > 0
+        state.tracers.ρχ = @SVector [FT(ii) for ii in 1:number_of_tracers]
+    end
+
+    nothing
+end
+
+function config_heldsuarez(FT, poly_order, resolution)
+    # Set up a reference state for linearization of equations
+    temp_profile_ref = DecayingTemperatureProfile{FT}(param_set)
+    ref_state = HydrostaticState(temp_profile_ref)
+
+    # Set up a Rayleigh sponge to dampen flow at the top of the domain
+    domain_height::FT = 30e3               # distance between surface and top of atmosphere (m)
+    z_sponge::FT = 20e3                    # height at which sponge begins (m)
+    α_relax::FT = 1 / 3600                 # sponge relaxation rate (1/s)
+    exp_sponge = 2                         # sponge exponent for squared-sinusoid profile
+    u_relax = SVector(FT(0), FT(0), FT(0)) # relaxation velocity (m/s)
+    sponge = RayleighSponge{FT}(
+        domain_height,
+        z_sponge,
+        α_relax,
+        u_relax,
+        exp_sponge,
+    )
+
+    # Set up the atmosphere model
+    exp_name = "HeldSuarez"
+    T_ref::FT = 255        # reference temperature for Held-Suarez forcing (K)
+    τ_hyper::FT = 8 * 3600 # hyperdiffusion time scale in (s)
+
+    if number_of_tracers > 0
+        δ_χ = @SVector [FT(ii) for ii in 1:number_of_tracers]
+        tracers = NTracers{number_of_tracers, FT}(δ_χ)
+    else
+        tracers = NoTracers()
+    end
+
+    model = AtmosModel{FT}(
+        AtmosGCMConfigType,
+        param_set;
+        ref_state = ref_state,
+        turbulence = ConstantViscosityWithDivergence(FT(0)),
+        hyperdiffusion = DryBiharmonic(τ_hyper),
+        moisture = DryModel(),
+        source = (Gravity(), Coriolis(), held_suarez_forcing!),
+        init_state_conservative = init_heldsuarez!,
+        data_config = HeldSuarezDataConfig(T_ref),
+        tracers = tracers,
+    )
+
+    config = ClimateMachine.AtmosGCMConfiguration(
+        exp_name,
+        poly_order,
+        resolution,
+        domain_height,
+        param_set,
+        init_heldsuarez!;
+        model = model,
+    )
+
+    return config
+end
+
+function held_suarez_forcing!(
```

Comment:
Can we abstract this out somewhere, so we do not need to duplicate code here and in heldsuarez.jl? I'd prefer for this forcing to be defined and coded in only one place.

Reply:
I agree. I don't know where we would put this very specific forcing function. Would need to create a separate space.

Reply:
Perhaps we just combine the two drivers into one.

Reply:
Done!